src/features/projectsV2/ProjectsContext.tsx

```diff
@@ -58,6 +62,9 @@ export const ProjectsProvider: FC<ProjectsProviderProps> = ({
   setSelectedMode,
 }) => {
   const [projects, setProjects] = useState<MapProject[] | null>(null);
+  const [singleProject, setSingleProject] = useState<ExtendedProject | null>(
+    null
+  );
```

Comment:
This does not get set to `null` while going back from a project. As a result, while opening another project details page, I initially see the UI of the old project. 

Note: see if there are other context values that should be unset while visiting or going back from a project.